### PR TITLE
DOP-3661: disallow duplicate changelog directives

### PR DESCRIPTION
**Pull request by mmeigs**
Emits error on duplicate `openapi_changelog` directives on a page.

Also, does legwork to account for `openapi_changelog` directives in `parser.py` and `postprocess.py`

### CODE REVIEW

File: snooty/parser.py [hunk 1]
```diff
@@ -712,6 +713,19 @@ def handle_directive(
                 )
                 return doc
 
+        elif name == "openapi-changelog":
+            # Version Changelog will be dependent on present api-version option
+            api_version = options.get("api-version", None)
+
+            if not argument_text == "cloud":
```
Comment (i80and) [inline snooty/parser.py:720]: this should be `if argument_text != "cloud"`

File: snooty/parser.py [hunk 2]
```diff
@@ -712,6 +713,19 @@ def handle_directive(
...
+                self.diagnostics.append(ExpectedPathArg(name, line))
+                return doc
+
+            if api_version:
+                return doc
+            else:
```
Comment (i80and) [inline snooty/parser.py:726]: No need to put this in an `else`, since if the condition is true we immediately return.

**Review by i80and — CHANGES_REQUESTED** [review]
looks good modulo two nits! we could make the diagnostic a little more helpful if you really want to dig into the parser a bit more, but I think the functionality is okay as is!

**Comment by mmeigs** [comment]
> we could make the diagnostic a little more helpful if you really want to dig into the parser a bit more

Good idea! I've added a couple diagnostics and tests to beef it up! Let me know what ya think

**Review by i80and — APPROVED** [review]
Delightful! lgtm! :shipit:

**Pull request merged**